drivers/i3c/i3c_mcux.c

```diff
@@ -140,10 +139,9 @@ static int reg32_poll_timeout(volatile uint32_t *reg,
 {
 	/*
 	 * These polling checks are typically satisfied
-	 * quickly (some sub-microseconds) so no extra
-	 * delay between checks.
+	 * quickly. Add a small delay between checks
 	 */
-	if (!WAIT_FOR((sys_read32((mm_reg_t)reg) & mask) == match, timeout_us, /*nop*/)) {
+	if (!WAIT_FOR((sys_read32((mm_reg_t)reg) & mask) == match, timeout_us, k_usleep(1))) {
```

Comment:
We're trading CPU load for latency, and technically it could happen with k_usleep() that FIFO buffer underflows.
I would recommend implementing this using interrupts instead removing the need for active polling.